torchnet/meter/apmeter.py

```diff
@@ -0,0 +1,135 @@
+    positively labeled, and smaller when the model believes the example should
+    be negatively labeled (for instance, the output of a sigmoid function); (2)
+    the `target` contains only values 0 (for negative examples) and 1
+    (for positive examples); and (3) the `weight` ( > 0) represents weight for
+    each sample.
+    """
+    def __init__(self):
+        self.reset()
+
+    def reset(self):
+        """Resets the meter with empty member variables"""
+        self.scores = torch.FloatTensor(torch.FloatStorage())
+        self.targets = torch.LongTensor(torch.LongStorage())
+        self.weights = torch.FloatTensor(torch.FloatStorage())
+
+    def add(self, output, target, weight=None):
+        """
+        Args:
+            output (Tensor): NxK tensor that for each of the N examples
+                indicates the probability of the example belonging to each of
+                the K classes, according to the model. The probabilities should
+                sum to one over all classes
+            target (Tensor): binary NxK tensort that encodes which of the K
+                classes are associated with the N-th input
+                    (eg: a row [0, 1, 0, 1] indicates that the example is
+                         associated with classes 2 and 4)
+            weight (optional, Tensor): Nx1 tensor representing the weight for
+                each example (each weight > 0)
+        """
+        if not torch.is_tensor(output):
+            output = torch.from_numpy(output)
+        if not torch.is_tensor(target):
+            target = torch.from_numpy(target)
+
+        if weight is not None:
+            if not torch.is_tensor(weight):
+                weight = torch.from_numpy(weight)
+            weight = weight.squeeze()
+        if output.dim() == 1:
+            output = output.view(output.size(0), 1)
+        else:
+            assert output.dim() == 2, \
+                'wrong output size (should be 1D or 2D with one column \
+                per class)'
+        if target.dim() == 1:
+            target = target.view(target.size(0), 1)
+        else:
+            assert target.dim() == 2, \
+                'wrong target size (should be 1D or 2D with one column \
+                per class)'
+        if weight is not None:
+            assert weight.dim() == 1, 'Weight dimension should be 1'
+            assert weight.numel() == target.size(0), \
+                'Weight dimension 1 should be the same as that of target'
+            assert torch.min(weight) >= 0, 'Weight should be non-negative only'
+        assert torch.equal(target**2, target), \
+            'targets should be binary (0 or 1)'
+        if self.scores.numel() > 0:
+            assert target.size(1) == self.targets.size(1), \
+                'dimensions for output should match previously added examples.'
+
+        # make sure storage is of sufficient size
+        if self.scores.storage().size() < self.scores.numel() + output.numel():
```

Comment:
That was mostly just to keep it consistent with the lua version.  The equivalent on the actual tensor would be `self.scores.numel()`, so in that case I could probably change the line to:  but we resize the storage of the tensor to contain